src/doc/unstable-book/src/library-features/asm.md

```diff
@@ -345,6 +345,29 @@ The `h` modifier will emit the register name for the high byte of that register
 
 If you use a smaller data type (e.g. `u16`) with an operand and forget the use template modifiers, the compiler will emit a warning and suggest the correct modifier to use.
 
+## Memory address operands
+
+Sometimes assembly instructions require operands passed via memory addresses/memory locations.
+You could wrap inputs/outputs in `[]` to indicate they as memory operands.
+
+For example:
+
+Old way:
+```rust,allow_fail
+# #![feature(llvm_asm)]
+# fn load_fpu_control_word(control: u16) {
+unsafe { llvm_asm!("fldcw $0" :: "m" (control) :: "volatile") }
+# }
+```
+
+New way:
+```rust,allow_fail
+# #![feature(asm)]
+# fn load_fpu_control_word(control: u16) {
+unsafe { asm!("fldcw [{:e}]", in(reg) control, options(nostack)) }
```

Comment:
```suggestion
unsafe { asm!("fldcw [{:e}]", in(reg) &control, options(nostack)) }
```
You want to pass a register containing the address of `control`, not `control` itself.

Reply:
What happens if one does not? Will it error or it will just UB?